austraits_build_tutorial.qmd

```diff
@@ -0,0 +1,625 @@
+austraits::summarise_austraits(austraits, var =  "family")
+austraits::summarise_austraits(austraits, var =  "genus")
+```
+
+Or just look at info for nitrogen-related traits:
+
+```{r, eval = FALSE}
+austraits::lookup_trait(austraits, "_N_") -> N_traits
+
+austraits$traits %>% 
+  dplyr::filter(trait_name %in% N_traits) %>% 
+  dplyr::distinct(taxon_name, trait_name) %>% 
+  dplyr::mutate(counter = 1) %>% 
+  dplyr::group_by(trait_name) %>% 
+    dplyr::mutate(counter = sum(counter)) %>% 
+    dplyr::distinct(trait_name, counter) %>% 
+  dplyr::ungroup()
+```
+
+Or look at how many datasets have each trait:
+
+```{r, eval = FALSE}
+austraits::lookup_trait(austraits, "_N_") -> N_traits
+
+austraits$traits %>% 
+  dplyr::filter(trait_name %in% N_traits) %>% 
+  dplyr::distinct(dataset_id, trait_name) %>% 
+  dplyr::mutate(counter = 1) %>% 
+  dplyr::group_by(trait_name) %>% 
+    dplyr::mutate(counter = sum(counter)) %>% 
+    dplyr::distinct(trait_name, counter) %>% 
+  dplyr::ungroup()
+```
+
+If you've sourced the file `extra_functions.R`, there are a few functions that allow you to summarise trait values.
+
+For instance, `categorical_summary` indicates how many times a specific trait value is reported for a given taxa (across all datasets):
+
+```{r, eval = FALSE}
+cat_summary <- categorical_summary(austraits, "resprouting_capacity")
+View(cat_summary)
+```
+
+Alternatively, create a wider matrix with possible trait values as columns:
+
+```{r, eval = FALSE}
+categorical_summary_wider <- 
+  categorical_summary_by_value(austraits, "resprouting_capacity") %>%
+    tidyr::pivot_wider(names_from = value_tmp, values_from = replicates)
+```
+
+We're hesitant to write functions to summarise numeric traits, as this quickly requires statistical assumptions.
+
+The datasets that comprise AusTraits were collected by different people, with different number of replicates, different entity types are reported - one dataset might include 20 measurements on individuals for a trait and another dataset might have submitted a single population-level mean derived from 5 measurements.
+
+How do you take the mean of these trait values?
+
+Do you want to include both data from experiments and plants growing under natural conditions? This information is recorded in the `basis_of_record` column.
+
+One algorithm we've developed is:
+
+```{r, eval = FALSE}
+weighted <- austraits_weighted_means(austraits, c("leaf_mass_per_area", "leaf_length"))
+```
+
+This function first groups values at the site level, then at the taxon level. For trait data sourced from floras, where trait values are documented as a minimum and maximum value, it simply takes the mean of these as an additional mean.
+
+## A complexity: pivoting datasets
+
+AusTraits datasets can be pivoted from `long` to `wide` format:
+
+```{r, eval = FALSE}
+Edwards_2000_values <-
+  austraits %>%
+     austraits::extract_dataset("Edwards_2000")
+
+Edwards_2000_pivoted <- 
```

Comment:
Nice idea to start with simple example!

Reply:
Oh, reading further, maybe this isn't a simple example. I'd suggest starting with aspire example (i..e single stingy top, not context, before getting more complex.

Reply:
I've included another example before it for `Farrell_2017` which is a dataset where all observations have entity_type = individual